src/core/Geometry.js

```diff
@@ -545,6 +545,58 @@ Geometry.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 	},
 
+	computeAngleVertexNormals: function ( angle ) {
+
+		function weightedNormal( normals, vector ) {
+
+		  var normal = new THREE.Vector3();
+
+			for ( var i = 0; i < normals.length; i ++ ) {
+
+				if ( normals[ i ].angleTo( vector ) < angle ) {
+
+			  		normal.add( normals[ i ] );
+
+				}
+
+			}
+
+		  return normal.normalize();
+
+		}
+
+		this.computeFaceNormals();
+
+		var vertexNormals = this.vertices.map( () => [] );
```

Comment:
Please avoid the usage of an array function.

Reply:
hm.. three.js not welcome with es6?